File: ccc08s2.py
import sys
from bisect import bisect_right

points = []
for i in range(25001):
	for j in range(25001):
		temp = (i ** 2 + j ** 2)*2-1
		points.append(temp)
		points.append(temp)

points.sort()
radius = int(sys.stdin.readline())
while radius != 0:
	print(bisect_right(points, radius))
	radius = int(sys.stdin.readline())

File: ccc08s2.cpp
#include <iostream>
#include <cmath>

int main() {
	int radius; std::cin >> radius;
	while (radius != 0) {
		int points = (radius*2+1)*(radius*2+1);
		for (int i = radius; i >= 1; i--) {
			for (int j = radius; j >= 1; j--) {
				if (std::sqrt(i*i + j*j) <= radius) {
					//std::cout << i << " " << j << "\n";
					break;
				}
				points -= 4;
				//std::cout << i << " " << j << " " << (float)std::sqrt(i*i + j*j) << '\n';
			}
		}
		std::cout << points << '\n';
		std::cin >> radius;
	}
	return 0;
}